### ISSUE TITLE
Token issue when signing in using "Portal for ArcGIS"

### ISSUE THREAD

**Issue by bsvensson** [Esri/ago-assistant#182]
After logging in using Portal for ArcGIS on https://ago-assistant.esri.com/, my Chrome console shows:
```
API v5.2
Uncaught (in promise) TypeError: Cannot read property 'length' of undefined
    at main.min.js:2
    at <anonymous>
```
And my folders do not show up.

I can see the `generateToken` request go to the correct portal. Ditto for the `self`request, but then there are request that seem to go to the wrong URL/portal: 
 https://www.arcgis.com/sharing/rest/community/users/tester1?token=MYTOKEN&f=json which returns `Invalid token` which makes sense, and if I search for things it also searches www.arcgis.com...

**Comment by ecaldwell** [comment]
What's the response on the `generateToken` request?

**Comment by bsvensson** [comment]
`{"token":"MYTOKEN","expires":1494541135122,"ssl":false}`

**Comment by ecaldwell** [comment]
Maybe it's not getting what it wants in the call to `self`. Is this a special dev portal where things might have changed? What version? Do you see properties in the `self` response like `isPortal` ? Relevant code [here](https://github.com/Esri/ago-assistant/blob/master/src/js/main.js#L195-L207).

**Comment by bsvensson** [comment]
@ecaldwell - I sent you a twitter DM with some details... :)

**Comment by ecaldwell** [comment]
I suspect it's failing the Portal test. Have you tried with a different user in the same Portal?

**Comment by bsvensson** [comment]
Yes, it works with an org user on the same portal.  I'll submit a PR with an easy fix.

**Comment by bsvensson** [comment]
Oops I referred to the wrong issue number in the commit https://github.com/Esri/ago-assistant/pull/183/commits/992c5f17ccbd1137a0066dcbf9292576b99140b2.
But PR - https://github.com/Esri/ago-assistant/pull/183 should be fine otherwise.

**Comment by ecaldwell** [comment]
Just to summarize, this issue only occurs on internal Esri dev environments where `isPortal` is `false` and `portalHostname` is something other than `www.arcgis.com`. The change in #183 resolves this without affecting non-Esri deployments.

Thanks for the quick debugging!

**Comment by hogpilot** [comment]
@bsvensson @ecaldwell @rwmajor2 Can someone confirm the status of this issue?  I recommend closing it since #183 was merged

**Comment by ecaldwell** [comment]
@hogpilot Thanks for the reminder. This was fixed in #183.
